sirepo_flyer.py

```diff
@@ -0,0 +1,241 @@
+            inputs = self.params_to_change[i]
+            for key, parameters_to_update in inputs.items():
+                elem_name.append(key)  # e.g., 'Aperture'
+                curr_param.append(list(parameters_to_update.keys()))  # e.g., 'horizontalSize'
+
+        return_dict = {self.name:
+                        {f'{self.name}_image': {'source': f'{self.name}_image',
+                                                'dtype': 'array',
+                                                'shape': [-1, -1],
+                                                'external': 'FILESTORE:'},
+                         f'{self.name}_shape': {'source': f'{self.name}_shape',
+                                               'dtype': 'array',
+                                               'shape': [1]},
+                         f'{self.name}_mean': {'source': f'{self.name}_mean',
+                                               'dtype': 'number',
+                                               'shape': []},
+                         f'{self.name}_photon_energy': {'source': f'{self.name}_photon_energy',
+                                               'dtype': 'number',
+                                               'shape': []},
+                         f'{self.name}_horizontal_extent': {'source': f'{self.name}_horizontal_extent',
+                                               'dtype': 'array',
+                                               'shape': [2]},
+                         f'{self.name}_vertical_extent': {'source': f'{self.name}_vertical_extent',
+                                               'dtype': 'array',
+                                               'shape': [2]},
+                         }
+                       }
+        for i in range(len(elem_name)):
+            for j in range(len(curr_param[i])):
+                return_dict[self.name][f'{self.name}_{elem_name[i]}_{curr_param[i][j]}'] = {'source': f'{self.name}_{elem_name[i]}_{curr_param[i][j]}',
+                                     'dtype': 'number',
+                                     'shape': []}
+        return return_dict
+
+    def collect(self):
+        # get results and clean up the copied simulations
+        shapes = []
+        means = []
+        photon_energies = []
+        horizontal_extents = []
+        vertical_extents = []
+        for i in range(len(self._copies)):
+            data_file = self._copies[i].get_datafile()
+            with open(self._srw_files[i], 'wb') as f:
+                f.write(data_file)
+
+            ret = read_srw_file(self._srw_files[i])
+            means.append(ret['mean'])
+            shapes.append(ret['shape'])
+            photon_energies.append(ret['photon_energy'])
+            horizontal_extents.append(ret['horizontal_extent'])
+            vertical_extents.append(ret['vertical_extent'])
+
+            print('copy {} data hash: {}'.format(self._copies[i].sim_id, hashlib.md5(data_file).hexdigest()))
```

Comment:
Let's keep these hash values and save them in databroker to validate the files later in the future.